src/main/kotlin/com/demonwav/mcdev/platform/mixin/util/Invoker.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Minecraft Dev for IntelliJ
+ *
+ * https://minecraftdev.org
+ *
+ * Copyright (c) 2019 minecraft-dev
+ *
+ * MIT License
+ */
+
+package com.demonwav.mcdev.platform.mixin.util
+
+import com.demonwav.mcdev.util.constantStringValue
+import com.demonwav.mcdev.util.findAnnotation
+import com.demonwav.mcdev.util.findMatchingMethod
+import com.demonwav.mcdev.util.ifEmpty
+import com.demonwav.mcdev.util.mapFirstNotNull
+import com.intellij.psi.PsiAnnotation
+import com.intellij.psi.PsiClass
+import com.intellij.psi.PsiMember
+import com.intellij.psi.PsiMethod
+import org.jetbrains.annotations.Contract
+
+@Contract(pure = true)
+fun PsiMember.findInvokerTarget(): PsiMember? {
+    val accessor = findAnnotation(MixinConstants.Annotations.INVOKER) ?: return null
+    val containingClass = containingClass ?: return null
+    val targetClasses = containingClass.mixinTargets.ifEmpty { return null }
+    return resolveInvokerTarget(accessor, targetClasses, this)
+}
+
+@Contract(pure = true)
+fun resolveInvokerTarget(
+        invoker: PsiAnnotation,
+        targetClasses: Collection<PsiClass>,
+        member: PsiMember
+): PsiMember? {
+    val value = invoker.findDeclaredAttributeValue("value")?.constantStringValue ?: return null
+    return when (member) {
```

Comment:
This can be an if-else expression.

Reply:
This is my first foray into kotlin so I made use of what I saw elsewhere in the project. This sounds more like a style issue unless the `return when` construct is inherently less efficient vs a typical if/else.

Reply:
Well, generally the point of a when expression is for cases with 3 or more branches.